code/game/gamemodes/game_mode.dm

```diff
@@ -400,15 +400,19 @@ proc/display_roundstart_logout_report()
 			to_chat(M, msg)
 
 
-proc/get_nt_opposed()
+/proc/get_nt_opposed()
 	var/list/dudes = list()
 	for(var/mob/living/carbon/human/man in player_list)
 		if(man.client)
+			//don't try picking someone like the captain or a security officer for potential collaborators, even if they ARE opposed to Nanotrasen for some reason

```

Comment:
There is nothing to stop the officer from RP-ing a dirty cop, this just prevents them from (as anoonki mentioned) getting free codewords and being approached by the traitor out of the blue.

If the cop wants to cut a deal with the traitor he's cornered/brigged, that's still possible. He's just gotta make that choice himself without any prompting.